numpy/polynomial/polyutils.py

```diff
@@ -661,10 +663,47 @@ def _fit(vander_f, x, y, deg, rcond=None, full=False, w=None):
         msg = "The fit may be poorly conditioned"
         warnings.warn(msg, RankWarning, stacklevel=2)
 
-    if full:
-        return c, [resids, rank, s, rcond]
-    else:
+    if not full and not cov:
         return c
+ 
+    return_tuple = [c]
+
+    if full:
+        return_tuple.append([resids, rank, s, rcond])
+    if cov:
+        """
+        v_base can be directly calculated as
+        >v_base = np.inv(np.dot(lhs, lhs.T))
+
+        However, this is numerically unstable.
+        It is calculated with SVD considering
+        the Vandermonde Matrix (lhs) as the
+        Jacobian Matrix.
+        """
+
+        U, s, _ = svd(lhs, full_matrices=False)
+
+        threshold = np.finfo(float).eps * max(lhs.shape) * s[0]
+        mask_array = s > threshold
+        s = s[mask_array]
+        U = U[:, mask_array]
+        v_base = np.dot(U / s**2, U.T)
+
+        if cov == "unscaled":
+            fac = 1
+        else:
+            if len(x) <= order:
```

Comment:
```
'''
Rescale the calculated covariance matrix by effectively coercing the reduced chi squared parameter to be unity. Note that this can result in inaccurate results when use with a poor fit model.
'''
```

Reply:
I tried to add this comment and I think that you are right. However, all checks fail if I split `cov `parameter into two (they did not even start). I think this is related with documentation and some function declaration throughout the project. The commit for this change was https://github.com/numpy/numpy/pull/20889/commits/39f51481d82d081866af072c8126da19a8e24702 which I reverted.

The best thing would be to make this change in another Pull Request and we can work on it. If we apply this change in this PR, we have to modify some tests and, as I said, more parts of the code. The neatest thing would be to open another PR.

Reply:
@lucasSaavedra123 interesting, I don't know how numpy's tests or anything are set up. But it's strange that modifications to the `cov` parameter would affect any existing tests since the function doesn't have the `cov` parameter now. Unless there are legacy tests related to the old polynomial code that touch this code? I don't like doing it in a separate PR because then the change from this PR to the next would be breaking. It seems like right now is the opportunity to make it however we think is best and then it's going to be stuck like that forever.

Reply:
OK. I'm going to figure out what is going on with the change I did.

Reply:
Uh... I understand the problem. It was IDE-related. Now it is solved.

Reply:
@jagerber48 Check if now it is correct my understanding of `absolute_w` that you detailed. As it is coded so far, `cov` and 'absolute_w' may be equal to `True `or `False`. `absolute_w=True` is equivalent to `cov="unscaled"`

Reply:
@lucasSaavedra123 Yes, the logic in the code is now comparable to the logic in `scipy` `curve_fit`. I think this is a good way to do it.